canopen/pdo/base.py

```diff
@@ -387,39 +388,38 @@ def save(self) -> None:
             logger.info("Setting SYNC start value to %d", self.sync_start_value)
             self.com_record[6].raw = self.sync_start_value
 
-        if self.map is not None:
-            try:
-                self.map_array[0].raw = 0
-            except SdoAbortedError:
-                # WORKAROUND for broken implementations: If the array has a
-                # fixed number of entries (count not writable), generate dummy
-                # mappings for an invalid object 0x0000:00 to overwrite any
-                # excess entries with all-zeros.
-                self._fill_map(self.map_array[0].raw)
-            subindex = 1
-            for var in self.map:
-                logger.info("Writing %s (0x%X:%d, %d bits) to PDO map",
-                            var.name, var.index, var.subindex, var.length)
-                if hasattr(self.pdo_node.node, "curtis_hack") and self.pdo_node.node.curtis_hack:  # Curtis HACK: mixed up field order
-                    self.map_array[subindex].raw = (var.index |
-                                                    var.subindex << 16 |
-                                                    var.length << 24)
-                else:
-                    self.map_array[subindex].raw = (var.index << 16 |
-                                                    var.subindex << 8 |
-                                                    var.length)
-                subindex += 1
-            try:
-                self.map_array[0].raw = len(self.map)
-            except SdoAbortedError as e:
-                # WORKAROUND for broken implementations: If the array
-                # number-of-entries parameter is not writable, we have already
-                # generated the required number of mappings above.
-                if e.code != 0x06010002:
-                    # Abort codes other than "Attempt to write a read-only
-                    # object" should still be reported.
-                    raise
-            self._update_data_size()
+        try:
+            self.map_array[0].raw = 0
+        except SdoAbortedError:
+            # WORKAROUND for broken implementations: If the array has a
+            # fixed number of entries (count not writable), generate dummy
+            # mappings for an invalid object 0x0000:00 to overwrite any
+            # excess entries with all-zeros.
+            self._fill_map(self.map_array[0].raw)
+        subindex = 1
+        for var in self.map:
+            logger.info("Writing %s (0x%X:%d, %d bits) to PDO map",
+                        var.name, var.index, var.subindex, var.length)
+            if getattr(self.pdo_node.node, "curtis_hack", False):  # Curtis HACK: mixed up field order
```

Comment:
```suggestion
            if getattr(self.pdo_node.node, "curtis_hack", False):
                # Curtis HACK: mixed up field order
```

Reply:
Well it is debatable, what the original Author wanted to do. Maybe we should initialize it with PDO_NOT_VALID or 0x0. But this would change the behavior even more. The value evaluation of this line

    self.com_record[1].raw = self.cob_id | PDO_NOT_VALID | (RTR_NOT_ALLOWED if not self.rtr_allowed else 0x0)

would return one of those if we ignore the fact you can't mix None with int type.

I also don't like silent fails, but I even more dislike broken default code. My Use-Case is the following: I have a lot of PDOs on devices, I have real device (aka master) on the bus, but for testing I fake some bus communication with my python canopen. So all my PDOs are already setup and configured by the real master. I want to use those PDOs without reading them all over the real bus, this triggers errors because the bus load is pretty high. Now I want to change some PDOs or verify (read) some PDOs without iterating over all devices and all PDOs to set the cob-id to 0x0.

Also your suggestion is mixed up I think.

Reply:
> Also your suggestion is mixed up I think.

This review comment's suggestion?  It just wraps the comment to the next line, avoiding line length warnings.  Just in the other place.